In [2]:
import pandas as pd

In [4]:
df = pd.read_csv("../data/wonka_choc_factory.csv")

In [5]:
df.head()  

Unnamed: 0  Order Date   Ship Date       Ship Mode  Customer ID  \
0           0  2021-03-31  2026-09-26  Standard Class       128055   
1           1  2021-03-31  2026-09-26  Standard Class       128055   
2           2  2021-09-15  2027-03-13  Standard Class       138100   
3           3  2021-09-15  2027-03-13  Standard Class       138100   
4           4  2022-10-04  2028-03-29     First Class       121391   

  Country/Region           City State/Province Postal Code   Division  \
0  United States  San Francisco     California       94122  Chocolate   
1  United States  San Francisco     California       94122  Chocolate   
2  United States  New York City       New York       10011  Chocolate   
3  United States  New York City       New York       10011  Chocolate   
4  United States  San Francisco     California       94109  Chocolate   

     Region                       Product Name  Sales  Units  Gross Profit  \
0   Pacific  Wonka Bar - Triple Dazzle Caramel   7.50      2          4.90   
1   Pacific     Wonka Bar -Scrumdiddlyumptious   7.20      2          5.00   
2  Atlantic          Wonka Bar - Fudge Mallows   7.20      2          4.80   
3  Atlantic         Wonka Bar - Milk Chocolate   9.75      3          6.33   
4   Pacific         Wonka Bar - Milk Chocolate   6.50      2          4.22   

   Cost          Factory   Latitude   Longitude  
0  2.60  Wicked Choccy's  32.076176  -81.088371  
1  2.20    Lot's O' Nuts  32.881893 -111.768036  
2  2.40    Lot's O' Nuts  32.881893 -111.768036  
3  3.42  Wicked Choccy's  32.076176  -81.088371  
4  2.28  Wicked Choccy's  32.076176  -81.088371

In [7]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10194 entries, 0 to 10193
Data columns (total 19 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   Unnamed: 0      10194 non-null  int64  
 1   Order Date      10194 non-null  object 
 2   Ship Date       10194 non-null  object 
 3   Ship Mode       10194 non-null  object 
 4   Customer ID     10194 non-null  int64  
 5   Country/Region  10194 non-null  object 
 6   City            10194 non-null  object 
 7   State/Province  10194 non-null  object 
 8   Postal Code     10194 non-null  object 
 9   Division        10194 non-null  object 
 10  Region          10194 non-null  object 
 11  Product Name    10194 non-null  object 
 12  Sales           10194 non-null  float64
 13  Units           10194 non-null  int64  
 14  Gross Profit    10194 non-null  float64
 15  Cost            10194 non-null  float64
 16  Factory         10194 non-null  object 
 17  Latitude        10194 non-null 

In [11]:
list(df.columns)

['Unnamed: 0',
 'Order Date',
 'Ship Date',
 'Ship Mode',
 'Customer ID',
 'Country/Region',
 'City',
 'State/Province',
 'Postal Code',
 'Division',
 'Region',
 'Product Name',
 'Sales',
 'Units',
 'Gross Profit',
 'Cost',
 'Factory',
 'Latitude',
 'Longitude']

In [12]:
df.columns = df.columns.str.replace(" ", "_").str.lower()

In [15]:
list(df.columns)

['unnamed:_0',
 'order_date',
 'ship_date',
 'ship_mode',
 'customer_id',
 'country/region',
 'city',
 'state/province',
 'postal_code',
 'division',
 'region',
 'product_name',
 'sales',
 'units',
 'gross_profit',
 'cost',
 'factory',
 'latitude',
 'longitude']

In [14]:
df["factory"].value_counts()   

Lot's O' Nuts        5692
Wicked Choccy's      4152
Secret Factory        217
The Other Factory     100
Sugar Shack            33
Name: factory, dtype: int64

In [ ]:
df["cost"].describe()

count    10194.000000
mean         4.742087
std          5.061647
min          0.600000
25%          2.400000
50%          3.600000
75%          5.700000
max        130.000000
Name: cost, dtype: float64

## ! the max is very high and interesting

In [27]:
df[df["cost"] == df["cost"].max()]

unnamed:_0  order_date   ship_date     ship_mode  customer_id  \
4379        4379  2021-04-13  2026-10-08  Second Class       122336   

     country/region          city state/province postal_code division  \
4379  United States  Philadelphia   Pennsylvania       19140    Other   

        region        product_name  sales  units  gross_profit   cost  \
4379  Atlantic  Lickable Wallpaper  260.0     13         130.0  130.0   

             factory   latitude  longitude  
4379  Secret Factory  41.446333 -90.565487

In [26]:
df['gross_profit'].describe()

count    10194.000000
mean         9.166451
std          6.643740
min          0.250000
25%          4.900000
50%          7.470000
75%         12.250000
max        130.000000
Name: gross_profit, dtype: float64

In [28]:
df['units'].describe()

count    10194.000000
mean         3.791838
std          2.228317
min          1.000000
25%          2.000000
50%          3.000000
75%          5.000000
max         14.000000
Name: units, dtype: float64

In [29]:
df[df['units'] == df['units'].max()]

unnamed:_0  order_date   ship_date       ship_mode  customer_id  \
146          146  2024-11-10  2030-05-04     First Class       155425   
228          228  2023-09-13  2029-03-10  Standard Class       105732   
263          263  2022-01-04  2027-07-02  Standard Class       104241   
553          553  2023-12-06  2029-06-01  Standard Class       103674   
1036        1036  2024-12-14  2030-06-10  Standard Class       151799   
1290        1290  2024-08-27  2030-02-17        Same Day       130036   
1480        1480  2023-10-27  2029-04-22  Standard Class       116596   
1558        1558  2022-08-24  2028-02-18  Standard Class       146563   
1647        1647  2024-06-26  2029-12-24  Standard Class       161410   
2202        2202  2022-10-02  2028-03-29    Second Class       119312   
2611        2611  2022-10-31  2028-04-26    Second Class       164448   
2745        2745  2021-02-17  2026-08-17  Standard Class       154165   
3954        3954  2022-09-19  2028-03-14    Second Class       135489   
4249        4249  2021-12-19  2027-06-13    Second Class       120768   
4734        4734  2024-10-30  2030-04-28  Standard Class       123491   
4902        4902  2024-11-24  2030-05-23  Standard Class       164028   
4930        4930  2024-01-01  2029-06-27  Standard Class       151750   
5220        5220  2022-03-21  2027-09-16  Standard Class       120621   
5225        5225  2021-03-11  2026-09-04    Second Class       158337   
5665        5665  2021-04-12  2026-10-08  Standard Class       154599   
5825        5825  2023-10-13  2029-04-10  Standard Class       145583   
6800        6800  2023-01-11  2028-07-07  Standard Class       141887   
6842        6842  2024-12-14  2030-06-10  Standard Class       169859   
8050        8050  2021-08-25  2027-02-17    Second Class       115259   
8764        8764  2023-06-11  2028-12-05  Standard Class       142405   
8911        8911  2023-03-15  2028-09-08  Standard Class       140571   
9042        9042  2024-10-12  2030-04-08  Standard Class       152702   
9203        9203  2022-07-24  2028-01-18  Standard Class       103135   
9434        9434  2021-12-27  2027-06-23  Standard Class       163447   
9624        9624  2022-09-18  2028-03-14    Second Class       149713   

     country/region           city  state/province postal_code   division  \
146   United States         Tucson         Arizona       85705  Chocolate   
228   United States          Omaha        Nebraska       68104  Chocolate   
263   United States     Alexandria        Virginia       22304  Chocolate   
553   United States    Los Angeles      California       90032  Chocolate   
1036         Canada        Toronto         Ontario         M3C  Chocolate   
1290  United States   Philadelphia    Pennsylvania       19120  Chocolate   
1480  United States  New York City        New York       10011  Chocolate   
1558  United States      Arlington           Texas       76017  Chocolate   
1647  United States   Philadelphia    Pennsylvania       19143  Chocolate   
2202  United States    Los Angeles      California       90045  Chocolate   
2611  United States        Salinas      California       93905  Chocolate   
2745  United States        Chicago        Illinois       60653  Chocolate   
3954  United States  New York City        New York       10024  Chocolate   
4249  United States       Florence         Alabama       35630  Chocolate   
4734  United States  San Francisco      California       94122  Chocolate   
4902  United States  San Francisco      California       94122  Chocolate   
4930  United States     Huntsville           Texas       77340  Chocolate   
5220  United States   Jacksonville  North Carolina       28540  Chocolate   
5225  United States  New York City        New York       10024  Chocolate   
5665  United States  Redondo Beach      California       90278  Chocolate   
5825  United States      Roseville      California       95661  Chocolate   
6800  United States       Columbus            Oh

In [30]:
df['sales'].describe()

count    10194.000000
mean        13.908537
std         11.341020
min          1.250000
25%          7.200000
50%         10.800000
75%         18.000000
max        260.000000
Name: sales, dtype: float64

# hypothesis options:
Red red: we are a marketing division doing research on:
1) where do we sell the most products? And what kind of products?
2) where do we have the most sales? And what do they buy?

So we can learn what type of marketing to use in what location.
For example, in one state or a county, a lot of kids by just one chocolate bar and in another state or county people by stacks (think about the pick order I called out above)

3. Personally, I am also interested in the order that was 260 in sales. Because it was 10 products, I would like to know what products and why they are so expensive and coming from the secret factory. (but I think this is more my personal curiosity and not relevant for the project)